source/onednngraph.js

```diff
@@ -0,0 +1,515 @@
+                            }
+                            else if (array != null) {
+                                array.push(number);
+                            }
+                        }
+                        if (array != null) {
+                            this._value = array;
+                        }
+                    }
+                    break;
+                case 'f32[]':
+                    if (this._value.length > 2 && this._value.startsWith('[') && this._value.endsWith(']')) {
+                        let array = [];
+                        const items = this._value.substring(1, this._value.length - 1).split(',')
+                            .map((item) => item.trim())
+                            .map((item) => item.endsWith('L') ? item.substring(0, item.length - 1) : item);
+                        for (const item of items) {
+                            number = Number.parseFloat(item);
+                            if (Number.isNaN(item - number)) {
+                                array = null;
+                            }
+                            else if (array != null) {
+                                array.push(number);
+                            }
+                        }
+                        if (array != null) {
+                            this._value = array;
+                        }
+                    }
+                    break;
+                default:
+                    throw new onednngraph.Error("Unsupported attribute type '" + schema.type + "'.");
+            }
+        }
+    }
+
+    get name() {
+        return this._name;
+    }
+
+    get type() {
+        return this._type;
+    }
+
+    get value() {
+        return this._value;
+    }
+
+    get visible() {
+        return this._visible == false ? false : true;
+    }
+};
+
+onednngraph.Parameter = class {
+
+    constructor(name, args) {
+        this._name = name;
+        this._arguments = args;
+    }
+
+    get name() {
+        return this._name;
+    }
+
+    get visible() {
+        return true;
+    }
+
+    get arguments() {
+        return this._arguments;
+    }
+};
+
+onednngraph.Argument = class {
+
+    constructor(name, type, initializer, lt) {
+        if (typeof name !== 'string') {
+            throw new onednngraph.Error("Invalid argument identifier '" + JSON.stringify(name) + "'.");
+        }
+        this._name = name;
+        this._type = type || null;
+        this._initializer = initializer || null;
+        this._lt = lt;
+    }
+
+    get name() {
+        return this._name;
+    }
+
+    get type() {
+        if (this._initializer) {
+            return this._initializer.type;
+        }
+        return this._type;
+    }
+
+    get initializer() {
+        return this._initializer;
+    }
+
+    get description() {
+        const stride = this._lt.stride ? ('[' + this._lt.stride.map((stride) => stride ? stride.toString() : '?').join(',') + ']') : '';
```

Comment:
Use `TensorType.denotation` or `Tensor.stride`. `description` should be an actual description provided by the model author, not a generated string.

Reply:
changed to stride. Any ideas for `layout_type(strided/any/undef)` and `property_type(constant/variable/undef)`?

Reply:
`constant` is `onednn.Argument` exposing `initializer`. The `initializer` is a `onednn.Tensor` object which exposes a `stride` property if the tensor is strided.

Reply:
got your point. `property_type ` in onednn means whether a tensor is constant. For example, given weight of  convolution(a.k.a input[1]), it can be `Variable` in a training graph and `constant` in a inference graph.